### PR TITLE
Mesh painting with ray tracing

### PR DESCRIPTION
**Pull request by Ceanze**
### Changes
* Paint mask textures are now rendered in mirrors.
* When creating an external resource in the render graph it is now possible to disable synchronization.

### CODE REVIEW

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp [hunk 1]
```diff
@@ -1071,16 +1077,65 @@ namespace LambdaEngine
 			m_MaterialInstanceCounts[materialIndex]++;
 		}
 
+		// Update resource for the entity mesh paint textures that is used for ray tracing
+		bool hasPaintMask = false;
+		if (m_RayTracingEnabled)
+		{
+			ECSCore* pECS = ECSCore::GetInstance();
+			const ComponentArray<MeshPaintComponent>* meshPaintComponents = pECS->GetComponentArray<MeshPaintComponent>();
```
Comment (Vonsen) [inline LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp]:
```suggestion
			const ComponentArray<MeshPaintComponent>* pMeshPaintComponents = pECS->GetComponentArray<MeshPaintComponent>();
```

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp [hunk 2]
```diff
@@ -1155,11 +1210,47 @@ namespace LambdaEngine
...
 			TArray<AccelerationStructureInstance>& asInstances = meshAndInstancesIt->second.ASInstances;
+			const uint32 textureIndex = asInstances[instanceIndex].CustomIndex & 0xFF;
 			asInstances[instanceIndex] = asInstances.GetBack();
 			asInstances.PopBack();
 			m_DirtyASInstanceBuffers.insert(&meshAndInstancesIt->second);
 			m_TLASDirty = true;
+
+			// Remove and reorder the paint mask textures (if needed) and set new indicies for ASInstances
+			ECSCore* pECS = ECSCore::GetInstance();
+			const ComponentArray<MeshPaintComponent>* meshPaintComponents = pECS->GetComponentArray<MeshPaintComponent>();
```
Comment (Vonsen) [inline LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp]:
```suggestion
			const ComponentArray<MeshPaintComponent>* pMeshPaintComponents = pECS->GetComponentArray<MeshPaintComponent>();
```

**Review by Vonsen — CHANGES_REQUESTED** [review]

**Review by Vonsen — APPROVED** [review]

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp (hunk 1 above)
Comment (Ceanze) [reply LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp]: ok :(

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp (hunk 2 above)
Comment (Ceanze) [reply LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp]: ok :(

**Pull request merged**